src/fs/rock/RockRebuild.cc

```diff
@@ -156,92 +203,138 @@ class LoadingParts
     /* Anti-padding storage. With millions of entries, padding matters! */
 
     /* indexed by sfileno */
-    std::vector<uint64_t> sizes; ///< LoadingEntry::size for all entries
-    std::vector<uint32_t> versions; ///< LoadingEntry::version for all entries
+    Ipc::Mem::Pointer<Sizes> sizes; ///< LoadingEntry::size for all entries
+    Ipc::Mem::Pointer<Versions> versions; ///< LoadingEntry::version for all entries
 
     /* indexed by SlotId */
-    std::vector<Ipc::StoreMapSliceId> mores; ///< LoadingSlot::more for all slots
+    Ipc::Mem::Pointer<Mores> mores; ///< LoadingSlot::more for all slots
 
     /* entry flags are indexed by sfileno; slot flags -- by SlotId */
-    std::vector<LoadingFlags> flags; ///< all LoadingEntry and LoadingSlot flags
+    Ipc::Mem::Pointer<Flags> flags; ///< all LoadingEntry and LoadingSlot flags
 };
 
 } /* namespace Rock */
 
+/* LoadingPartsOwner */
+
+Rock::LoadingPartsOwner::LoadingPartsOwner(const SwapDir *dir):
+    sizes(shm_new(Sizes)(SizesPath(dir->path).c_str(), dir->entryLimitActual())),
+    versions(shm_new(Versions)(VersionsPath(dir->path).c_str(), dir->entryLimitActual())),
+    mores(shm_new(Mores)(MoresPath(dir->path).c_str(), dir->slotLimitActual())),
+    flags(shm_new(Flags)(FlagsPath(dir->path).c_str(), dir->slotLimitActual()))
+{
+    auto moresArray = mores->object();
+    for (int i = 0; i < moresArray->capacity; ++i)
+        moresArray->items[i] = -1;
+}
+
```

Comment:
I am probably missing something important here. Why not add a second LoadingPartsOwner(dir, rebuild) constructor that would call shm_old, keeping the contents of the old segments intact? That second constructor can assert that its `rebuild` parameter is always true.

LoadingParts can naturally get all the segments (old or new -- they do not care) from `partsOwner`. In fact, I am not even sure that the LoadingParts class would be needed at all in this case.

AFAICT, that way, we can always initialize and always delete `partsOwner` and LoadingPartsOwner destructor can always unlink the segments, making all the UnlinkNamed()-related changes (and possibly even LoadingParts polishing) unnecessary.